### ISSUE TITLE
The CHANGELOG.md file hasn't been updated in 9 months.

### ISSUE THREAD

**Issue by devonhk** [argoproj/argo-cd#13799]
Checklist:

* [ x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x ] I've included steps to reproduce the bug.
* [ x] I've pasted the output of `argocd version`.

**Describe the bug**

The changelog file is no longer being updated.

**To Reproduce**

https://github.com/argoproj/argo-cd/blob/master/CHANGELOG.md#v248-2022-07-29

**Expected behavior**

The changelog file should be updated on every release. Github releases aren't a replacement to a changelog file IMO, as it's much easier to grep for breaking changes in a single file then have to open dozens of browser tabs to go through each release.

**Comment by crenshaw-dev** [comment]
Yep. We'll need a one-time update and then some automation added to automatically update the changelog.

**Comment by cdemarco-drw** [comment]
This is in fact a bug, and should be labeled as such.

The CHANGELOG is unexpectedly out-of-date and does not point the reader to the Releases UI.

Worse, it does not contain a warning that it has become out-of-date, so the unaware reader will think they have the current changelog. *Incorrect information is worse than missing information.*

This is an unconscionable regression to a released product, it should be prioritized near the top of the list.

**Comment by slmg** [comment]
In terms of user experience browsing git history / diff is not feasible to figure out what is *worth knowing* - https://github.com/argoproj/argo-cd/compare/v2.8.4...v2.10.5

Please consider updating the CHANGELOG.md manually until a better process is put in place. Given the size of this project thousands of people must be left scratching their heads between releases, trending toward two years without changes update.

In terms of future automation, this could help: https://towncrier.readthedocs.io/en/stable/

Thank you for the amazing work.

**Comment by crenshaw-dev** [comment]
We've seen a positive change in terms of the number of people cutting releases (yay!) but a side effect is that we've had less standardization in terms of how/when people do changelog updates.

I think the only really sustainable strategy is automation, which effectively means fixing the release notes generator (goreleaser) and adding a changelog updater. Unfortunately I don't think either of those is high enough on any maintainer's priority list to happen soon. If anyone from the community can help, that would be very appreciated!

**Comment by Zebradil** [comment]
@crenshaw-dev could you briefly explain (or point to the corresponding docs if they exist) what the release flow looks like? I might be able to help.

I personally prefer using https://github.com/googleapis/release-please/ with Goreleaser. Here is an [example workflow](https://github.com/mykso/myks/blob/8429049d58de38b85c022ea8a2b087b10088d489/.github/workflows/flow-stage.yml) that updates a "release PR" when something gets merged into the main branch. It uses conventional commits and semver to build a list of changes for the changelog. When the PR is merged, Goreleaser proceeds with building and publishing.
